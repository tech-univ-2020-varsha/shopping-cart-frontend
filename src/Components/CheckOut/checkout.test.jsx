import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';

import { Router } from 'react-router-dom';
import routeData from 'react-router';
import { createMemoryHistory } from 'history';
import axios from 'axios';
import Checkout from './checkout';

describe('The Checkout component', () => {
  const mockLocation = {
    pathname: '/welcome',
    hash: '',
    search: '',
    state: '',
  };
  beforeEach(() => {
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
  });
  const mockAxios = jest.spyOn(axios, 'get');
  mockAxios.mockResolvedValue({ data: [] });
  it('should render correctly', () => {
    const history = createMemoryHistory();
    const { asFragment } = render(<Router history={history}><Checkout /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should make an api call to fetch cart details', () => {
    const history = createMemoryHistory();
    act(() => {
      const { asFragment } = render(<Router history={history}><Checkout /></Router>);
      expect(mockAxios).toHaveBeenCalled();
    });
  });
});
