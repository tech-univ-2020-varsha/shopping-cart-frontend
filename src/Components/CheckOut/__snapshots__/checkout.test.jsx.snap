// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Checkout component should render correctly 1`] = `
<DocumentFragment>
  <div>
    Loading!...
  </div>
</DocumentFragment>
`;
